packages/frontend/src/pages/campaigns/filters/checkboxes-filter/index.tsx

```diff
@@ -0,0 +1,96 @@
+import { Template } from "@carrot-kpi/sdk";
+import { Checkbox, Typography } from "@carrot-kpi/ui";
+import React, { ChangeEvent, useCallback, useEffect, useState } from "react";
+import { FiltersLoading } from "../loading";
+import {
+    CheckboxesObj,
+    generateGroupId,
+    getDefaultCheckboxes,
+    getDefaultSelected,
+} from "../utils";
+
+interface CheckboxesFilterProps {
+    items: Template[];
+    groupId: string;
+    title: string;
+    loading: boolean;
+    selected: Set<number>;
+    setSelected: (newSelected: Set<number>) => void;
+    defaultChecked: (item: Template) => boolean;
+}
+
+export const CheckboxesFilter = ({
+    items,
+    title,
+    loading,
+    selected,
+    setSelected,
+    defaultChecked,
+    groupId,
+}: CheckboxesFilterProps) => {
+    const [checkboxes, setCheckboxes] = useState<CheckboxesObj>({});
+
+    const initializeCheckboxes = useCallback(() => {
+        const hasItems = items.length !== 0;
+        const hasCheckboxes = Object.keys(checkboxes).length !== 0;
+
+        if (hasItems && !hasCheckboxes) {
+            const defaultCheckboxes = getDefaultCheckboxes(
+                items,
+                defaultChecked
+            );
+            setCheckboxes(defaultCheckboxes);
+            setSelected(getDefaultSelected(defaultCheckboxes));
+        }
+    }, [checkboxes, defaultChecked, items, setSelected]);
+
+    const handleCheckedChange = (e: ChangeEvent<HTMLInputElement>) => {
+        const element = e.target;
+        const elementId = Number(e.target.getAttribute("data-id"));
```

Comment:
Here we can use `element`, so `element.getAttribute(...)`

Reply:
Yes. Thaaaanks

Reply:
fixed 0b0e3127244ed8bb7ede963d6eb0868f403cc73a